client/src/pages/search/SearchComparision.js

```diff
@@ -0,0 +1,149 @@
+import React, { useState, useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { setConfig } from "../../slice/configLoadSlice";
+import { Sidebar } from "./../../layouts/Sidebar";
+import { Layout } from "./../../layouts/Layout";
+import { ModelSelect } from "../../components/modelSelect";
+import { FileSelectComponent } from "../../components/fileSelect";
+// import { socket } from "../../socket";
+
+import { io } from "socket.io-client";
+
+export function ComparisionPage() {
+  // Declare a new state variable, which we'll call "count"
+  const dispatch = useDispatch();
+
+  let serverUrl = "localhost:3000";
+  let socket = io(serverUrl);
+
+  const [isConnected, setIsConnected] = useState(socket.connected);
+  const [benchmark, setBenchmark] = useState({});
+  const [benchmark2, setBenchmark2] = useState({});
+
+  React.useEffect(() => {
+    dispatch(setConfig());
+    socket.connect();
+    const bencmarkListiner = (message) => {
+      console.log("coming here benchmark");
+      setBenchmark(message);
+    };
+    const bencmarkListiner2 = (message) => {
+      console.log("coming here benchmark");
+      setBenchmark2(message);
+    };
+
+    function onConnect() {
+      console.log("connected");
+      setIsConnected(true);
+    }
+    socket.on("connect", onConnect);
+    socket.on("benchmark", bencmarkListiner);
```

Comment:
Minor but "bencmark" is misspelled --> should be "benchmark"! 
In other places (ex. lines 31, 32) we use "benchmark", so just stylistically if making that change isn't too difficult 😅